sw/device/silicon_creator/rom_ext/rom_ext_start.S

```diff
@@ -208,14 +237,121 @@ _rom_ext_start_boot:
    */
   la gp, __global_pointer$
 
-  // Re-enable linker relaxation.
-  .option pop
-
   /**
-   * Jump to C Code
+   * Jump to C Code using `jr` to make sure we are jumping to the virtual address.
```

Comment:
Ideally I think we'd make this transition more explicit but I'm not sure how unfortunately. This approach seems a bit fragile. Something we should bear in mind.